drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -224,9 +224,17 @@ int i2c_stm32_target_register(const struct device *dev,
 
 	if (!data->slave_cfg) {
 		data->slave_cfg = config;
+	if( data->slave_cfg->flags == I2C_SLAVE_FLAGS_ADDR_10_BITS )
```

Comment:
Since https://github.com/zephyrproject-rtos/zephyr/pull/43166, you should now use I2C_TARGET_FLAGS_ADDR_10_BITS

Reply:
Fixed